internal/cmd/marketplace/apply.go

```diff
@@ -0,0 +1,319 @@
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+const (
+	applyLong = `Create or update one or more Marketplace items.
+
+You can either specify:
+    - one or more files, with the flag -f 
+    - one or more directories, with the flag -d
+	
+Supported formats are JSON (.json files) and YAML (.yaml or .yml files).`
+
+	applyExample = `
+# Apply the configuration of the file myFantasticGoTemplate.json located in the current directory to the Marketplace
+miactl marketplace apply -f myFantasticGoTemplate.json
+
+# Apply the configurations in myFantasticGoTemplate.json and myFantasticNodeTemplate.yml to the Marketplace, with relative paths
+miactl marketplace apply -f ./path/to/myFantasticGoTemplate.json -f ./path/to/myFantasticNodeTemplate.yml
+
+# Apply all the valid configuration files in the directory myFantasticGoTemplates to the Marketplace
+miactl marketplace apply -d myFantasticGoTemplates`
+)
+
+// ApplyCmd returns a new cobra command for adding or updating marketplace resources
+func ApplyCmd(options *clioptions.CLIOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "apply { { -f file-path }... | { -d directory-path }... }",
+		Short:   "Create or update Marketplace items",
+		Long:    applyLong,
+		Example: applyExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if options.MarketplaceResourcesDirPath == "" && len(options.MarketplaceResourceFilePaths) == 0 {
+				return errors.New(`at least one of  "directory" or "file" must be set`)
+			}
+
+			restConfig, err := options.ToRESTConfig()
+			cobra.CheckErr(err)
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+
+			var resourceFilesPaths []string
+			if len(options.MarketplaceResourceFilePaths) > 0 {
+				resourceFilesPaths = options.MarketplaceResourceFilePaths
+			}
+			if options.MarketplaceResourcesDirPath != "" {
+				resourceFilesPaths, err = buildPathsListFromDir(options.MarketplaceResourcesDirPath)
+				cobra.CheckErr(err)
+			}
+			applyReq, err := buildApplyRequest(resourceFilesPaths)
+			cobra.CheckErr(err)
+			outcome, err := applyMarketplaceResource(cmd.Context(), client, restConfig.CompanyID, applyReq)
+			cobra.CheckErr(err)
+
+			fmt.Println(buildOutcomeSummaryAsTables(outcome))
+
+			return nil
+		},
+	}
+
+	options.AddMarketplaceApplyFlags(cmd.Flags())
+	cmd.MarkFlagsMutuallyExclusive("file", "directory")
+
+	return cmd
+}
+
+const applyEndpoint = "/api/backend/marketplace/tenants/%s/resources"
+
+var (
+	errResWithoutName       = errors.New(`the required field "name" was not found in the resource`)
+	errNoValidFilesProvided = errors.New("no valid files were provided, see errors above")
+	errDuplicatedResName    = errors.New("some resources have duplicated name field")
+	errResNameNotAString    = errors.New(`the field "name" must be a string`)
+	errInvalidExtension     = errors.New("file has an invalid extension. Valid extensions are `.json`, `.yaml` and `.yml`")
```

Comment:
Why are we using ' / " / `?

Reply:
Because there are double quotes inside some strings, and I like it more than adding escape characters